src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -258,4 +272,8 @@ protected int doHashCode() {
     public String getWriteableName() {
         return NAME;
     }
+
+    private static boolean isClusterOnOrAfterMinRequiredVersion() {
```

Comment:
One problem with this is that the function name does not indicate that it has anything to do with default model ID. I would recommend parametrizing this with the version.